### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by dpyskunova**
Need to update backend for /search_post later and add an ability to open post

### CODE REVIEW

File: src/components/Common/Button/FbLoginButton/FbLoginButton.tsx [hunk 1]
```diff
@@ -26,17 +26,19 @@ export class FbLoginButton extends Component<IFacebookRegisterProps, {}> {
       location: (response.location && response.location.name) || ''
     };
 
-    auth(userDetails).then(response => {
+    auth('/user_register', userDetails).then(response => {
       const token = response.token;
```
Comment (fstodulski) [inline src/components/Common/Button/FbLoginButton/FbLoginButton.tsx]: try `const {token} = response;`

File: src/components/Common/Button/FbLoginButton/FbLoginButton.tsx [hunk 2]
```diff
@@ -26,17 +26,19 @@ export class FbLoginButton extends Component<IFacebookRegisterProps, {}> {
...
+      token && localStorage.setItem('token', token);
+      response.sprededResponse && localStorage.setItem(
         'userDetails',
         JSON.stringify(response.sprededResponse)
       );
       this.props.stores.userDetails.user = {
         ...response.sprededResponse,
         token
       };
+    }).catch(function(error) {
```
Comment (fstodulski) [inline src/components/Common/Button/FbLoginButton/FbLoginButton.tsx]: `.catch((error:Error)=> console.log(error))`

File: src/components/Login/Login.tsx [hunk 3]
```diff
@@ -54,15 +54,13 @@ export class Login extends React.Component <ILoginProps, ILoginState> {
   };
 
   authenticate = (userDetails:any) => {
-    auth(userDetails).then(response => {
-      if (response.sprededResponse.password && !(encrypt(response.sprededResponse.password, 10) === encrypt(userDetails.password, 10))) {
+    auth('/user_login', userDetails).then(response => {
         const token = response.token;
```
Comment (fstodulski) [inline src/components/Login/Login.tsx]: `const {token} = response;`

File: src/components/Login/Login.tsx [hunk 4]
```diff
@@ -54,15 +54,13 @@ export class Login extends React.Component <ILoginProps, ILoginState> {
...
-        localStorage.setItem('userDetails', JSON.stringify(response.sprededResponse));
-        localStorage.setItem('token', response.token);
-        this.props.stores.userDetails.user  =  {...response.sprededResponse, token};
-      } else {
-        console.log('The username and password did not match.');
-      }
+        response.foundedUser && localStorage.setItem('userDetails', JSON.stringify(response.foundedUser));
+        token && localStorage.setItem('token', response.token);
+        this.props.stores.userDetails.user = {...response.foundedUser, token};
+    }).catch(function(error) {
```
Comment (fstodulski) [inline src/components/Login/Login.tsx]: `.catch((error:Error) => console.log(error));`

File: src/components/Navigation/Navigation.scss [hunk 5]
```diff
@@ -4,6 +4,6 @@ header {
 
 .navigation-grid {
   display: grid;
-  grid-template-columns: 1fr 1fr;
+  grid-template-columns: 1fr 1fr 1fr;
```
Comment (fstodulski) [inline src/components/Navigation/Navigation.scss]: `grid-template-columns: repeat(3, 1fr);`

File: src/components/Search/Search.scss [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+.react-autosuggest__suggestions-container--open {
+  display: block;
+  position: absolute;
+  top: 31px;
+  width: 280px;
+  border: 1px solid #aaa;
+  background-color: #fff;
+  font-family: Helvetica, sans-serif;
```
Comment (fstodulski) [inline src/components/Search/Search.scss]: Font-familly is declared in App.scss, no need to duplicate it

File: src/components/Search/Search.scss [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  position: relative;
+  display: grid;
+  justify-items: end;
+}
+
+.react-autosuggest__input {
+  width: 240px;
+  height: 10px;
+  padding: 10px 20px;
+  font-family: Helvetica, sans-serif;
```
Comment (fstodulski) [inline src/components/Search/Search.scss]: Font-familly is declared in App.scss, no need to duplicate it

File: src/components/Search/Search.tsx [hunk 8]
```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
```
Comment (fstodulski) [inline src/components/Search/Search.tsx:1]: Please change order of imports:
1. tsx
2. mobx
4. interfaces
5. scss

File: src/components/Search/Search.tsx [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+  stores?: any;
+}
+
+@inject('stores')
+@observer
+export class Search extends Component<ISearchProps> {
+  constructor(props: ISearchProps) {
+    super(props);
+
+    this.state = {
```
Comment (fstodulski) [inline src/components/Search/Search.tsx]: If you use State please create interface for it

**Review by fstodulski — CHANGES_REQUESTED** [review]

File: package.json [hunk 10]
```diff
@@ -70,11 +73,14 @@
...
   "devDependencies": {
     "@types/jest": "^23.3.1",
+    "@types/material-ui": "^0.21.5",
     "@types/node": "^10.9.2",
     "@types/react": "^16.4.12",
     "@types/react-dom": "^16.0.7",
     "@types/react-facebook-login": "^3.6.5",
     "@types/react-router-dom": "^4.3.0",
+    "@types/react-tap-event-plugin": "^0.0.30",
+    "material-ui": "^0.20.2",
```
Comment (stanimirtt) [inline package.json]: Why we need material ui?

File: src/components/Search/Search.tsx (hunk 9 above)
Comment (stanimirtt) [reply src/components/Search/Search.tsx]: It is not good practice to mix mobx with state.

File: package.json [hunk 11]
```diff
@@ -71,11 +73,13 @@
   },
   "devDependencies": {
     "@types/jest": "^23.3.1",
+    "@types/material-ui": "^0.21.5",
```
Comment (suezzo) [inline package.json]: We don't use it

**Review by anastasiialisikh — APPROVED** [review]

**Review by suezzo — APPROVED** [review]

File: src/components/Search/Search.tsx [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+      {suggestion.name}
+    </div>
+  );
+
+  getSuggestions = (value:any) => {
+    const inputValue = value.trim().toLowerCase();
+    const inputLength = inputValue.length;
+    const { tweets } = this.state;
+
+    return inputLength === 0 ? [] : tweets.filter((lang:any) =>
```
Comment (stanimirtt) [inline src/components/Search/Search.tsx]: inputLength === 0 ->  inputValue.length

File: package.json (hunk 11 above)
Comment (dpyskunova) [reply package.json]: Deleted material-ui with all dependencies

**Review by stodorov-ss — APPROVED** [review]

**Pull request merged**